### ISSUE TITLE
Clarification in number of layers

### ISSUE THREAD

**Issue by amitness** [pytorch/fairseq#927]
In the paper, it mentions that we have 24 layers in the large model. Why does the len(all_layers) return 25?

![image](https://user-images.githubusercontent.com/8587189/62113355-2bd98f80-b2d4-11e9-9490-ba8d414102fc.png)

**Comment by myleott** [comment]
The first is the embedding lookup + positional embeddings, followed by 24 transformer blocks. You can see how the `inner_states` list is populated here: https://github.com/pytorch/fairseq/blob/master/fairseq/modules/transformer_sentence_encoder.py#L207-L214

**Comment by amitness** [comment]
@myleott Thank you. It makes sense now.

I had another query. Assuming a toy example of a spam/ham classifier, if I want to finetune the roberta model, is this the correct way?
1. Add new classification head with num_classes=1 and name 'spam'
2. Set requires_grad = False for all the pretrained model layers
3. Enable gradients in this spam head block
4. Train
![image](https://user-images.githubusercontent.com/8587189/62131264-900f4a00-b2fa-11e9-8e88-420dfc3a7b20.png)

**Comment by myleott** [comment]
In that case you should set `num_classes=2` for `spam` and `ham`.

You can leave `requires_grad=True` everywhere if you want to finetune the whole RoBERTa network. Otherwise you can set `requires_grad=False` if you want to freeze portions of the network, but typically people get better results finetuning the entire network (i.e. leaving `requires_grad=True`).

**Comment by amitness** [comment]
@myleott Thanks a lot.
